src/cascadia/WpfTerminalControl/TerminalContainer.cs

```diff
@@ -275,10 +277,11 @@ internal void RaiseResizedIfDrawSpaceIncreased()
         /// <inheritdoc/>
         protected override void OnDpiChanged(DpiScale oldDpi, DpiScale newDpi)
         {
-            if (this.terminal != IntPtr.Zero)
+            if (this.terminal == IntPtr.Zero)
             {
-                NativeMethods.TerminalDpiChanged(this.terminal, (int)(NativeMethods.USER_DEFAULT_SCREEN_DPI * newDpi.DpiScaleX));
+                return;
             }
+            NativeMethods.TerminalDpiChanged(this.terminal, (int)(NativeMethods.USER_DEFAULT_SCREEN_DPI * newDpi.DpiScaleX));

```

Comment:
I would prefer the previous version here. The early return doesn't help in this case.